DEVELOPMENT.md

```diff
@@ -1,72 +1,167 @@
+   ```
+7. Click yes if a popup suggests switching to the virtual environment. Otherwise, go through these steps:
+	  1. Open any python file in the directory.
+	  2. Select the interpreter selector on the blue status bar at the bottom of the editor.
+    <img src="/docs/source/_static/img/vscode-status-bar.png" alt="VS Code Status Bar"></img>
+	  3. Switch to the path that includes .venv from the dropdown at the top.
+    <img src="/docs/source/_static/img/vscode-select-venv.png" alt="VS Code Interpreter Selection Menu"></img>
+
+8. Update your PowerShell execution policies. Win+x followed by the 'a' key opens the admin Windows PowerShell. Enter the following command to allow the virtual environment activation script to run:
+   ```
+   Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
+   ```
+</details>
+
+## Making Changes
+
+<details><summary><h3>using the Command Line</h3></summary>
+
+1. Make sure you're on the main branch.
+   ```bash
+   git switch main
+   ```
 
+2. Use the git pull command to retrieve content from the BentoML Github repository.
+   ```bash
+   git pull
+   ```
+
+3. Create a new branch and switch to it.
+   ```bash
+   git switch -c my-new-branch-name
+   ```
+
+4. Make your changes!
+
+5. Use the git add command to save the state of files you have changed.
+   ```bash
+   git add <names of the files you have changed>
+   ```
+
+6. Commit your changes.
+   ```bash
+   git commit
+   ```
+
+7. Push all changes to your fork on GitHub.
+   ```bash
+   git push
+   ```
+</details>
 
-### Creating Bento with custom bentoml source
+<details><summary><h3>using VS Code</h3></summary>
+
+1. Switch to the main branch:
+    1. Open the command palette with Ctrl+Shift+P.
+    2. Search for 'Git: Checkout to...'
+    3. Select 'main'.
 
-It is possible to force a Bento to use a custom BentoML source distribution:
+2. Pull from the upstream remote:
+    1. Open the command palette.
+    2. Enter and select 'Git: Pull...'
+    3. Select 'upstream'.
 
-1. Install custom BentoML in editable mode. e.g.:
-   * git clone your bentoml fork
-   * `pip install -e PATH_TO_THE_FORK`
-2. Set env var `export BENTOML_BUNDLE_LOCAL_BUILD=True`
-   * make sure you have the latest setuptools installed:  `pip install -U setuptools`
-3. Build a new Bento with `bentoml build` in your project directory
-4. The new Bento will include a wheel file built from the BentoML source, and 
-`bentoml containrize` will install it to override the default BentoML installation in base image
+3. Create and change to a new branch:
+    1. Type in 'Git: Create Branch...' in the command palette.
+    2. Enter a branch name.
 
+4. Make your changes!
+
+5. Stage all your changes:
+    1. Enter and select 'Git: Stage All Changes...' in the command palette.
+
+6. Commit your changes:
+    1. Open the command palette and enter 'Git: Commit'.
+
+7. Push your changes:
+    1. Enter and select 'Git: Push...' in the command palette.
+
+</details>
```

Comment:
I'm abit hesistant to add this. What is your opinion on this @parano ? I think if we provide a way to do it in the terminal then using VSCode should be pretty self-explanatory.

Reply:
I like this addition, this could be very helpful for beginners